astroid/raw_building.py

```diff
@@ -319,65 +319,73 @@ def ast_from_function(func, built_objects, module, name=None, parent=None):
+            default = node_classes.Empty()
+            annotation = node_classes.Empty()
             if parameter.default is not _Parameter.empty:
-                defaults.extend(_ast_from_object(parameter.default, built_objects, module, parent=parent))
+                default = _ast_from_object(parameter.default, built_objects,
+                                           module, parent=parent)
+
             if parameter.annotation is not _Parameter.empty:
-                annotations.extend(_ast_from_object(parameter.annotation, built_objects, module, parent=parent))
-            else:
-                annotations.append(None)
-        return names, defaults, annotations
+                annotation = _ast_from_object(parameter.annotation, built_objects,
+                                              module, parent=parent)
 
-    def extract_vararg(parameter):
-        '''Takes a single-element iterator possibly containing a Parameter and
-        returns a name and an annotation.
+            param.postinit(default=default, annotation=annotation)
+            yield param
 
-        '''
+    def _extract_vararg(parameter, parent):
+        """Build a variadic parameter from the given Python parameter."""
         try:
-            return parameter[0].name
+            parameter = parameter[0]
         except IndexError:
-            return None
-
-    vararg = parameters.get(_Parameter.VAR_POSITIONAL, ())
-    kwarg = parameters.get(_Parameter.VAR_KEYWORD, ())
-    vararg_name = extract_vararg(vararg)
-    kwarg_name = extract_vararg(kwarg)
-    args_node = node_classes.Arguments(vararg=vararg_name, kwarg=kwarg_name, parent=func_node)
-
-    # This ignores POSITIONAL_ONLY args, because they only appear in
-    # functions implemented in C and can't be mimicked by any Python
-    # function.
-    names, defaults, annotations = extract_args(parameters.get(_Parameter.POSITIONAL_OR_KEYWORD, ()), args_node)
-    kwonlynames, kw_defaults, kwonly_annotations = extract_args(parameters.get(_Parameter.KEYWORD_ONLY, ()), args_node)
-    args = [node_classes.AssignName(name=n, parent=args_node) for n in names]
-    kwonlyargs = [node_classes.AssignName(name=n, parent=args_node) for n in kwonlynames]
-    if vararg_name and vararg[0].annotation is not _Parameter.empty:
-        varargannotation = vararg.annotation
-    else:
-        varargannotation = None
-    if kwarg_name and kwarg[0].annotation is not _Parameter.empty:
-        kwargannotation = kwarg.annotation
-    else:
-        kwargannotation = None
+            return node_classes.Empty()
+
+        positional_only = parameter.kind == _Parameter.POSITIONAL_ONLY
+        if parameter.annotation is not _Parameter.empty:
+            annotation = _ast_from_object(parameter.annotation,
+                                          built_objects, module, parent=parent)[0]
+        else:
+            annotation = node_classes.Empty()
+
+        # TODO: check if it is okay to use the parent here?
```

Comment:
This will hopefully be irrelevant soon, but I don't understand this comment, so you should probably expand it.